### ISSUE TITLE
Invalid new array initialisation

### ISSUE THREAD

**Issue by kse300489** [zephir-lang/zephir#1159]
Example:

``` php
var
  variable;

let
  variable = 1;

... // any operations with variable

let
  variable = [variable];

var_dump(variable);
```

Output:

```
array(1) {
  [0]=>
  *RECURSION*
}
```

This happens because zephir generate invalid C code. 

``` c
ZEPHIR_INIT_NVAR(variable);
zephir_create_array(variable, 1, 0 TSRMLS_CC);
ZEPHIR_CHECK_POINTER(variable);
zephir_array_fast_append(variable, variable);
```

What it do:
1 Create new array and assign it to `variable`

``` php
let
  variable = [];
```

2 Add `variable` to `variable`. Add self to self.

``` php
let
  variable = [variable];
```

What he must do:
1 Create new array `tmp`

``` php
let
  tmp = [];
```

2 Add `variable` to `tmp`

``` php
let
  tmp[] = variable;
```

3 Assign `tmp` to `variable`

``` php
let
  variable = tmp;
```

4 Destroy `tmp`

``` php
unset(tmp);
```

**Comment by kse300489** [comment]
Ping?

**Comment by mruz** [comment]
I can confirm this bug, also this works:

``` zep
var tmp = variable;
let variable = [tmp];
```

**Comment by sergeyklay** [comment]
@dreamsxin Could you please take a look

**Comment by dreamsxin** [comment]
@sergeyklay Yes sir.

**Comment by sergeyklay** [comment]
Fixed in the [0.11.11 version](https://github.com/phalcon/zephir/blob/development/CHANGELOG.md#01111---2019-02-26). Feel free to open a new issue if the problem appears again. Thank you for contributing.
